accounts-db/src/accounts_cache.rs

```diff
@@ -213,7 +213,7 @@ impl AccountsCache {
             self
                 .cache
                 .entry(slot)
-                .or_insert(self.new_inner())
+                .or_insert_with(|| self.new_inner())
```

Comment:
I think you're doing or_insert to minimize the amount of insns you execute with the write lock held. But DashMap::default() does heap allocate, unlike HashMap::default(), so I don't think you want to do it. A better optimization here would be to set a capacity for the map, so that it does fewer reallocations once you do slot_cache.insert()